### PR TITLE
[4.0] Installer Checks

### PR DESCRIPTION
**Pull request by brianteeman**
This PR attempts to add error message explanations on the failing pre-install checks

I have also taken the opportunity to rename some of the constants used in the precheck and move them to the correct section.

I am not **100% confident** on the technical merit of the messages

### CODE REVIEW

File: installation/language/en-GB/en-GB.ini [hunk 1]
```diff
@@ -18,14 +18,28 @@ INSTL_WARNJAVASCRIPT="Warning! JavaScript must be enabled for proper installatio
...
+INSTL_DATABASE_SUPPORT="Database Support:"
+INSTL_JSON_SUPPORT_AVAILABLE="JSON Support"
+INSTL_MB_LANGUAGE_IS_DEFAULT="MB Language is Default"
+INSTL_MB_STRING_OVERLOAD_OFF="MB String Overload Off"
+INSTL_NOTICE_DATABASE_SUPPORT="No supported databases were found."
+INSTL_NOTICE_JSON_SUPPORT_AVAILABLE="Your PHP installation needs to have JSON enabled for Joomla to be installed!"
+INSTL_NOTICE_MBLANG_NOTDEFAULT="PHP mbstring language is not set to neutral. This can be set locally by entering <strong>php_value mbstring.language neutral</strong> in your <code>.htaccess</code> file."
+INSTL_NOTICE_MBSTRING_OVERLOAD_OFF="PHP mbstring function overload is set. This can be turned off locally by entering <strong>php_value mbstring.func_overload 0</strong> in your <code>.htaccess</code> file."
+INSTL_NOTICE_PARSE_INI_FILE_AVAILABLE="The required php functions <code>parse_ini_file</code> and <code>parse_ini_string</code> are disabled on your server."
+INSTL_NOTICE_XML_SUPPORT="XML Support is not available. This should be enabvled by default in php but Ubuntu users, may need to install by doing <code>sudo apt-get install php-xml</code> followed by web server restart."
```
Comment (ggppdk) [inline installation/language/en-GB/en-GB.ini]: enabled instead of enabvled ?

File: installation/language/en-US/en-US.ini [hunk 2]
```diff
@@ -18,14 +18,28 @@ INSTL_WARNJAVASCRIPT="Warning! JavaScript must be enabled for proper installatio
...
+INSTL_DATABASE_SUPPORT="Database Support:"
+INSTL_JSON_SUPPORT_AVAILABLE="JSON Support"
+INSTL_MB_LANGUAGE_IS_DEFAULT="MB Language is Default"
+INSTL_MB_STRING_OVERLOAD_OFF="MB String Overload Off"
+INSTL_NOTICE_DATABASE_SUPPORT="No supported databases were found."
+INSTL_NOTICE_JSON_SUPPORT_AVAILABLE="Your PHP installation needs to have JSON enabled for Joomla to be installed!"
+INSTL_NOTICE_MBLANG_NOTDEFAULT="PHP mbstring language is not set to neutral. This can be set locally by entering <strong>php_value mbstring.language neutral</strong> in your <code>.htaccess</code> file."
+INSTL_NOTICE_MBSTRING_OVERLOAD_OFF="PHP mbstring function overload is set. This can be turned off locally by entering <strong>php_value mbstring.func_overload 0</strong> in your <code>.htaccess</code> file."
+INSTL_NOTICE_PARSE_INI_FILE_AVAILABLE="The required php functions <code>parse_ini_file</code> and <code>parse_ini_string</code> are disabled on your server."
+INSTL_NOTICE_XML_SUPPORT="XML Support is not available. This should be enabvled by default in php but Ubuntu users, may need to install by doing <code>sudo apt-get install php-xml</code> followed by web server restart."
```
Comment (ggppdk) [inline installation/language/en-US/en-US.ini]: enabled instead of enabvled ?

**Comment by rdeutz** [comment]
For whatever reason system test are failing on this PR, this needs to be fixed before.
/cc @yvesh @puneet0191

File: installation/src/Model/ChecksModel.php [hunk 3]
```diff
@@ -69,14 +69,14 @@ public function getPhpOptions()
 		$option = new \stdClass;
 		$option->label  = \JText::_('INSTL_ZLIB_COMPRESSION_SUPPORT');
 		$option->state  = extension_loaded('zlib');
-		$option->notice = null;
+		$option->state ? null : \JText::_('INSTL_NOTICE_ZLIB_COMPRESSION_SUPPORT');
```
Comment (Quy) [inline installation/src/Model/ChecksModel.php]: Is this right?

Comment (brianteeman) [reply installation/src/Model/ChecksModel.php]: no ;)

**Comment by puneet0191** [comment]
okay, I will take a look at it tomorrow @rdeutz , breaks in the first step while selecting the language

**Comment by brianteeman** [comment]
@puneet0191 any news?

**Comment by puneet0191** [comment]
hey @brianteeman I am not sure what I am missing, I tried testing the PR on my local, I have Json enabled, but I am still getting the pre-installation check JSON Support Message. 
I think we have the same problem on System Test Docker Image. 
screenshot from 2018-06-19 12-41-01
screenshot from 2018-06-19 12-41-16

**Comment by brianteeman** [comment]
ah ok let me double check - maybe I left some of the testing code in there.

**Comment by brianteeman** [comment]
yes - its my fault -sorry to bother you - but at least it proves the tests work

Fixing it now

**Comment by puneet0191** [comment]
Thanks :+1:

**Comment by brianteeman** [comment]
Silly me - I left some changes I had to make to see the errors in the PR. All fixed now and ready for testing

**Comment by puneet0191** [comment]
I must say, what a wonderful work with system testing  :stuck_out_tongue:

**Review by dgrammatiko — APPROVED** [review]

**Comment by wilsonge** [comment]
Looks good to me! Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**